### PR TITLE
Dsns 109

### PR DESCRIPTION
**Pull request by dunkgray**
The main part is adding the code to reprocess new ls9 l1. 
To get this going I also refactored some of the code.
Plus there are tests.

### CODE REVIEW

File: scripts/examples/go_reprocess.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# ARD_ENV="/g/data/v10/projects/c3_ard/dea-ard-scene-select/scripts/prod/ard_env/prod-wagl-s2.env"
+
+ARD_ENV="/g/data/v10/projects/c3_ard/dea-ard-scene-select/scripts/prod/ard_env/prod-wagl-ls.env"
+
+# The module scene-select
+# time ard-scene-select --workdir scratch/  --pkgdir  scratch/ --logdir scratch/ --project u46 --walltime 10:00:00  \
+# --env $ARD_ENV  --products $PRODUCTS --scene-limit 999999 #--find-blocked
+
+# Scene select in this repo
+time python3 ../../scene_select/ard_reprocessed_l1s.py --workdir scratch/  --pkgdir  scratch/ --logdir scratch/ --project u46  \
```
Comment (gordonyeong) [inline scripts/examples/go_reprocess.sh:27]: might be better to consider being able to run ard_reprocessed_l1s.py from anywhere in the file system. One application is for automated testing.

`
script_directory=$(dirname $(dirname "$(readlink -f "$0")"))
test_data_directory="$script_directory/test_data/integration_tests"
`

File: tests/ard_reprocessed_l1s/check_db.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ODCCONF="--config ${USER}_dev.conf"
+  
+else
+  echo "not NCI"
+  ODCCONF="--config ${USER}_local.conf"
+  TEST_DATA=$HOME/test_data
+  # datacube -v  $ODCCONF system init
+fi
+
+datacube  $ODCCONF product list #
```
Comment (gordonyeong) [inline tests/ard_reprocessed_l1s/check_db.sh:23]: remove stray #

File: tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        "moved", "ga_ls9c_ard_3", "102", "076", "2022", "06", "27"
+    )
+    fname = new_dir.joinpath(
+        "ga_ls9c_ard_3-2-1_102076_2022-06-27_final.odc-metadata.yaml"
+    )
+    assert os.path.isfile(fname) == True
+
+    # uuids have been written to an archive file
+    filename = jobdir.joinpath(ARCHIVE_FILE)
+    temp = open(filename, "r").read().splitlines()
```
Comment (gordonyeong) [inline tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py:111]: hmmm.... prolly should use `with` when dealing with an input resource to signal the fact the contents are being used and sourced from the said input resource

Comment (gordonyeong) [reply tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py:111]: hmmmm perhaps better to explicitly specify encoding

File: tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    b_dir = REPROCESS_TEST_DIR.joinpath(
+        "l1_Landsat_C2",
+        "102_076",
+        "LC91020762022178",
+        "LC09_L1TP_102076_20220627_20220802_02_T1.tar",
+    )
+    assert sorted([str(a_dir), str(b_dir)]) == sorted(temp)
+    # There is a run ard pbs file
+    filename = jobdir.joinpath(PBS_ARD_FILE)
+    assert os.path.isfile(fname) == True
```
Comment (gordonyeong) [inline tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py:133]: comparison for boolean is to be using `is` not `==`.

File: scripts/examples/go_reprocess.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [[ $HOSTNAME == *"gadi"* ]]; then
+	echo If module load breaks check out a clean environment
```
[the remark this answers is not in the document]
Comment (dunkgray) [reply scripts/examples/go_reprocess.sh:4]: done

File: scripts/examples/go_reprocess.sh (hunk 1 above)
Comment (dunkgray) [reply scripts/examples/go_reprocess.sh]: set up so it can run from anywhere

File: tests/ard_reprocessed_l1s/check_db.sh (hunk 2 above)
Comment (dunkgray) [reply tests/ard_reprocessed_l1s/check_db.sh]: done

File: tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py (hunk 3 above)
Comment (dunkgray) [reply tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py]: done, thanks

File: tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py (hunk 4 above)
Comment (dunkgray) [reply tests/ard_reprocessed_l1s/test_ard_reprocessed_l1s.py]: done

File: modules/go.sh [hunk 6]
```diff
@@ -10,5 +10,5 @@ if [[ $1 == "--prod" ]]; then
     ./package-module.sh "${2:-$(date +'%Y%m%d')}" --prod
 else
     # dev version
-    ./package-module.sh  dev_${2:-$(date +'%Y%m%d')}
+    ./package-module.sh  dev_${2:-$(date +'%Y%m%d')} ---not-prod
```
Comment (gordonyeong) [inline modules/go.sh:13]: nice :D

**Review by gordonyeong — APPROVED** [review]

**Review by gordonyeong — APPROVED** [review]

**Pull request merged**